kadmin/sarama_kadmin.go

```diff
@@ -19,6 +21,56 @@ type SaramaKafkaAdmin struct {
 	sra      sradmin.SrAdmin
 }
 
+func (s *SaramaKafkaAdmin) GetClusterConfig() (ClusterConfig, error) {
+	MaybeIntroduceLatency()
+	log.Info("Attempting to describe Kafka cluster...")
+	saramaBrokers, _, err := s.admin.DescribeCluster()
```

Comment:
The actual IO calls should happen inside a go routine. Often I wrap it in a method starting with `do`, here it would be `go doGetClusterConfig(clusterChan, errChan)` where you pass in the result channels.